src/plugin.ts

```diff
@@ -1,236 +1,247 @@
+
+                // expand out @scoped plugins
+                relativePluginPaths.slice()
+                    .filter(path => path.charAt(0) === "@") // is it a scope directory?
+                    .forEach(scopeDirectory => {
+                        const index = relativePluginPaths.indexOf(scopeDirectory); // remove scopeDirectory from the path list
+                        relativePluginPaths.splice(index, 1);
+
+                        const absolutePath = path.join(searchPath, scopeDirectory);
+                        fs.readdirSync(absolutePath)
+                            .filter(name => fs.statSync(path.resolve(absolutePath, name)).isDirectory())
+                            .forEach(name => relativePluginPaths.push(name));
+                    });
+
+                relativePluginPaths
+                    .filter(relativePath => {
+                        const name = path.basename(relativePath); // we could have a dirname like "@somescope/homebridge-my-plugin"
+                        return PluginManager.isQualifiedPluginName(name);
+                    })
+                    .forEach(relativePath => {
+                        try {
+                            const absolutePath = path.resolve(searchPath, relativePath);
+                            const plugin = PluginManager.loadPlugin(absolutePath);
+
+                            plugins.push(plugin);
+                        } catch (error) {
+                            log.warn(error.message);
+                            return;
+                        }
+                    });
+            }
         });
-      }
+
+        return plugins;
     }
 
-    // does this path point inside a single plugin and not a directory containing plugins?
-    if (fs.existsSync(path.join(requirePath, "package.json")))
-      names = [""];
-
-    // read through each directory in this node_modules folder
-    for (var index2 in names) {
-      var name = names[index2];
-
-      // reconstruct full path
-      var pluginPath = path.join(requirePath, name);
-      try {
-        // we only care about directories
-        if (!fs.statSync(pluginPath).isDirectory()) continue;
-      } catch (e) {
-        continue;
-      }
-      // does this module contain a package.json?
-      var pjson;
-      try {
-        // throws an Error if this isn't a homebridge plugin
-        pjson = Plugin.loadPackageJSON(pluginPath);
-      }
-      catch (err) {
-        // is this "trying" to be a homebridge plugin? if so let you know what went wrong.
-        if (!name || name.indexOf('homebridge-') === 0 || (name.charAt(0) === '@' && name.split(path.sep).length > 1 && name.split(path.sep)[1].indexOf('homebridge-') === 0)) {
-          log.warn(err.message);
+    private static loadPlugin(absolutePath: string): Plugin {
+        const packageJson: PackageJSON = PluginManager.loadPackageJSON(absolutePath);
+        const name = PluginManager.extractPluginName(packageJson.name);
```

Comment:
Does this register the plugin in homebridge without the `@scope/`? If so we should be keeping the `@scope` so two plugins with the same name, but different scopes can be identified easily.

Reply:
Yeah it does actually. I tried to scrap the scope as soon as possible to try to make the code more understandable and more clear what is meant when the code refers to "pluginName".

Should this actually be a feature? I suspected a plugin with the same name but a different scope to just be the same plugin published in an organization(?). Also current hombridge code is not able to handle that either: https://github.com/homebridge/homebridge/blob/e67630a3978aa126786711f8ec06610db517ca74/lib/server.js#L166 (would register the plugin correctly, would just be overwritten in the dic maintained by the server [which is never used though, to be fair]).

So to get the change clear, two homebridge plugin with the same name but different scope should be treated as two different plugins? Will change that later.

Reply:
> So to get the change clear, two homebridge plugin with the same name but different scope should be treated as two different plugins? 

Yes, they should be treated as seperate plugins. This is so people can publish their own versions of existing plugins to test / fork, or just publish under their own scope if they feel like it. By keeping the scope in place the logs will also show exactly which plugin got loaded which will make things easier to troubleshoot when users send us their logs.

>  Also current hombridge code is not able to handle that either:

`plugin.name()` will return the `@oznu/homebridge-example` in the current codebase:

https://github.com/homebridge/homebridge/blob/e67630a3978aa126786711f8ec06610db517ca74/lib/plugin.js#L26-L28

I think just using exactly what is in the `name` attribute of the `package.json` is probably reasonable and simplest way to go. Thoughts?

Reply:
Lol I did read it the other way round 😅😄

Reply:
> I think just using exactly what is in the name attribute of the package.json is probably reasonable and simplest way to go. Thoughts?

Yeah probably. I just add a try catch in the server when we call the initializer. Should there be a conflict we want to fail gracefully.

Reply:
> Should there be a conflict we want to fail gracefully.

Sounds good. That's what the existing code does if it finds the same plugin in two different paths.

Reply:
Behavior was restored in latest force push.